### PR TITLE
ISPN-12684 Lazy start SearchMapping for protobuf caches

### PR DESCRIPTION
**Pull request by gustavonalle**
https://issues.redhat.com/browse/ISPN-12684

### CODE REVIEW

**Comment by gustavonalle** [comment]
Marking as preview to let CI run

**Comment by gustavonalle** [comment]
Fixed tests, ready for review

File: remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/LazySearchMapping.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+         searchMapping = builder != null ? builder.build() : null;
+      }
+      if (indexingConfiguration.enabled()) {
+         if (valueDataConversion.getStorageMediaType().match(APPLICATION_PROTOSTREAM)) {
+            // Try to resolve the indexed type names to protobuf type names.
+            Set<String> knownTypes = protobufMetadataManager.getSerializationContext().getGenericDescriptors().keySet();
+            for (String typeName : indexedEntityTypes) {
+               if (!knownTypes.contains(typeName)) {
+                  if (searchMapping != null) searchMapping.close();
+                  throw new CacheConfigurationException("The declared indexed type '" + typeName + "' is not known. Please register its proto schema file first.");
```
Comment (tristantarrant) [inline remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/LazySearchMapping.java:134]: I'd prefer an i18n exception here

**Review by tristantarrant — CHANGES_REQUESTED** [review]
Just one minor change

File: remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/LazySearchMapping.java (hunk 1 above)
Comment (gustavonalle) [reply remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/LazySearchMapping.java]: fixed!

**Comment by tristantarrant** [comment]
Merged ,thanks

**Pull request merged**